hazelcast/src/main/java/com/hazelcast/internal/util/SimpleCompletedFuture.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.util;
+
+import com.hazelcast.core.ExecutionCallback;
+import com.hazelcast.spi.InternalCompletableFuture;
+
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+
+import static com.hazelcast.util.ExceptionUtil.sneakyThrow;
+
+/**
+ * Generic implementation of a completed {@link InternalCompletableFuture}.
+ */
+public class SimpleCompletedFuture<E> implements InternalCompletableFuture<E> {
```

Comment:
@viliam-durina 
We had `CompletedFuture`. It looks like doing similar thing . Was there a reason for not using it ?

Reply:
That one isn't independent impl. It requires `SerializationService`, `Executor`. I wanted something simple and independent, similar to java8's `java.util.concurrent.CompletableFuture`.

In `ICompletableFuture` there's a comment about using java8's `CompletableFuture` in the future, but I think this will be by no means straightforward...

Reply:
When needed we have returned `CompletedFuture` so far when needed from all async API's. 
And their behaviour is different in some cases. 
For example , `andThen(callback)` runs synchronously in your case. But it is running on an internal executor in case of `CompletedFuture`. This may lead to unexpected results for our users.

Reply:
IMO, if user uses `andThen` without an Executor, he should be fine with the code running on any thread and should be quick enough. If he wants more control, he should use the version with  custom Executor, so that can control on which thread will it run...  This is what java8's CompletableFuture does as well, I think `ICompletableFutere` is it's backport (kind of). This seems to me as solving the problem in the wrong place. 

Also, we run it synchronously only in specific case (empty key set). Other times, it's run on other threads.